### PR TITLE
LIBCLOUD-761 network and monitoring support

### PR DESCRIPTION
**Pull request by tonybaloney**
Extended Dimension Data driver for compute to enable: 
- Creating, deleting and renaming "networks" MCP1 only
- Enabling, updating and disabling server monitoring (MCP 1 or 2)

All functions include tests. this is intended for the 0.20 release

### CODE REVIEW

File: libcloud/compute/drivers/dimensiondata.py [hunk 1]
```diff
@@ -419,7 +463,84 @@ def ex_list_networks(self, location=None):
...
+        """
+        Delete a network from an MCP 1 data center
+
+        :param  network: The network to delete
+        :type   network: :class:`DimensionDataNetwork`
+
+        :rtype: ``bool``
+        """
+        response = self.connection.request_with_orgId_api_1(
+            'network/%s?delete' % network.id,
```
Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py:511]: For query parameters, please use `params=` argument which takes care of correctly encoding query params for you.

File: libcloud/compute/drivers/dimensiondata.py [hunk 2]
```diff
@@ -419,7 +463,84 @@ def ex_list_networks(self, location=None):
...
+
+        :param  new_name: The new name of the network
+        :type   new_name: ``str``
+
+        :rtype: ``bool``
+        """
+        response = self.connection.request_with_orgId_api_1(
+            'network/%s' % network.id,
+            method='POST',
+            data='name=%s' % new_name).object
```
Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py:531]: Hm so payload in this request is neither xml or json, but just form encoded data?

**Comment by Kami** [comment]
Thanks.

Besides the params thing, looks good to me so far.

As far as the networking goes, we should also look into reviving this - https://github.com/apache/libcloud/pull/345

File: libcloud/compute/drivers/dimensiondata.py (hunk 2 above)
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py:531]: Yes. Inconsistent with the other APIs I know but they had to maintain back compat

File: libcloud/compute/drivers/dimensiondata.py (hunk 1 above)
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py:511]: You can still do that when the param doesn't have a matching value? I'll try that out

Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py:511]: Ah, that's a good point - I'm not actually sure you can do that.

**Comment by tonybaloney** [comment]
+1 on #345

**Comment by tonybaloney** [comment]
all good to merge+close?

**Comment by Kami** [comment]
LGTM, +1 - feel free to merge it.

**Comment by tonybaloney** [comment]
merged.

**Pull request merged**